### PR TITLE
Feature/v1-247 | Removed API adapters.

### PR DESCRIPTION
**Pull request by dapodev**
# Changes

1. Removed `adapterSender`, `adapterUser`, `adapterManager` and `clientCourseAdapter`
2. Refactored types
3. Refactored controllers

### CODE REVIEW

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/routing/pendingCourses.ts [hunk 1]
```diff
@@ -21,20 +19,16 @@ pendingCoursesRouter.get(
...
   // sendMail,
-  adapterSender,
 );
 pendingCoursesRouter.put(
   SubRoutes.declinePendingCourse,
   withAuth([USER_ROLES.MANAGER]),
-  adapterManager,
   declinePendingCourse,
   addNotification,
   // sendMail,
```
[the remark this answers is not in the document]
Comment (dapodev) [reply backend/src/routing/pendingCourses.ts:31]: Oh, I didn't even checked this out, cause we're not using it anywhere. Will we?

File: backend/src/routing/clientCoursesRouter.ts [hunk 2]
```diff
@@ -83,7 +78,6 @@ clientCoursesRouter.put(
   withAuth([USER_ROLES.MANAGER]),
   manageAssessment,
   getAchievements,
```
[the remark this answers is not in the document]
Comment (dapodev) [reply backend/src/routing/clientCoursesRouter.ts:80]: `getAchievements` not only preparing data to send to the client, but also **changes user skills** in db. And we're using this controller exactly for the purpose of "assigning achievements".

File: backend/src/controllers/pendingCourses/approvePendingCourse.ts [hunk 3]
```diff
@@ -45,8 +47,11 @@ const approvePendingCourse = async (
       await arrangeAssessment(clientCourseId);
     }
 
-    results.updateStatus = `Course was approved ${withAssessment ? 'with' : 'without'} assessment.`;
     next();
+
+    res.json({
+      updateStatus: `Course was approved ${withAssessment ? 'with' : 'without'} assessment.`,
```
Comment (SaXaPhonist) [inline backend/src/controllers/pendingCourses/approvePendingCourse.ts:53]: as you wrote  we don't need "update status", just plane string in res

File: backend/src/controllers/pendingCourses/declinePendingCourse.ts [hunk 4]
```diff
@@ -39,10 +41,11 @@ const declinePendingCourse = async (
     );
     await removeFromPendingFieldCourses(manager, clientCourse._id);
 
-    results.updateStatus = 'Course was declined.';
     await updateClientCourseField(clientCourseId, CLIENT_COURSE_FIELDS.applyDate, Date.now());
 
     next();
+
+    res.json({ updateStatus: 'Course was declined.' });
```
Comment (SaXaPhonist) [inline backend/src/controllers/pendingCourses/declinePendingCourse.ts:48]: need plane string

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/controllers/pendingCourses/approvePendingCourse.ts (hunk 3 above)
Comment (zhuk-vezunchikz) [reply backend/src/controllers/pendingCourses/approvePendingCourse.ts]: done

File: backend/src/controllers/pendingCourses/declinePendingCourse.ts (hunk 4 above)
Comment (zhuk-vezunchikz) [reply backend/src/controllers/pendingCourses/declinePendingCourse.ts]: done

**Pull request merged**